salt/states/ssh_auth.py

```diff
@@ -63,6 +63,13 @@
 import re
 import sys
 
+sshre = re.compile(
+    r"((?P<options>.*?)\s)?"
+    r"(?P<keytype>\S+?-\S+?|ed25519|ecdsa)\s"
```

Comment:
`ed25519` and `ecdsa` aren't listed in any man pages I looked at, but are in the documentation for these states. If they are not valid in any supported `sshd` versions then they could be removed and the docs changed. All other keytypes have at least one `-` in them.